### ISSUE TITLE
Effekt can't easily initialize arrays within arrays?

### ISSUE THREAD

**Issue by TheDying0fLight** [TheDying0fLight/FairSweeper#3]
The following will create an array filled with arrays which all have the same reference making it impossible to edit a single "cell" as it will change the whole column in the array.
`val arr: Array[Array[Int]] = array(10, array(10, 1))`

**Comment by jiribenes** [comment]
Yes, this is a consequence of the values vs computations distinction. :)
There's also the `build` "constructor" that takes an initial computation, which means you can do `array::build(10) { _ => array(10, 1) }`.

**Comment by TheDying0fLight** [comment]
Ah thanks didn't see the `build` function now I can reduce my code a bit 👍
